src/main/java/org/gridsuite/modification/server/modifications/byfilter/AbstractModificationByAssignment.java

```diff
@@ -0,0 +1,369 @@
+                .withUntypedValue(VALUE_KEY_EQUIPMENT_IDS, equipmentIds)
+                .withSeverity(TypedValue.WARN_SEVERITY)
+                .build());
+        }
+    }
+
+    private void applyModification(Identifiable<?> equipment,
+                                     AbstractAssignmentInfos abstractAssignmentInfos,
+                                     List<ReportNode> reports,
+                                     List<String> notEditableEquipments) {
+
+        // check pre-conditions
+        if (!preCheckValue(equipment, abstractAssignmentInfos, reports, notEditableEquipments)) {
+            return;
+        }
+
+        // perform to apply new value
+        try {
+            final String oldValue = getOldValue(equipment, abstractAssignmentInfos);
+            final String newValue = applyValue(equipment, abstractAssignmentInfos);
+            reports.add(ReportNode.newRootReportNode()
+                .withMessageTemplate(REPORT_KEY_EQUIPMENT_MODIFIED_REPORT,
+                    "        ${" + VALUE_KEY_EQUIPMENT_TYPE + "} id : ${" + VALUE_KEY_EQUIPMENT_NAME +
+                    "}, ${" + VALUE_KEY_FIELD_NAME + "} : ${" + VALUE_KEY_OLD_VALUE + "} → ${" + VALUE_KEY_NEW_VALUE + "}")
+                .withUntypedValue(VALUE_KEY_EQUIPMENT_TYPE, equipment.getType().name())
+                .withUntypedValue(VALUE_KEY_EQUIPMENT_NAME, equipment.getId())
+                .withUntypedValue(VALUE_KEY_FIELD_NAME, getEditedFieldLabel(abstractAssignmentInfos))
+                .withUntypedValue(VALUE_KEY_OLD_VALUE, oldValue == null ? NO_VALUE : oldValue)
+                .withUntypedValue(VALUE_KEY_NEW_VALUE, newValue)
+                .withSeverity(TypedValue.TRACE_SEVERITY)
+                .build());
+        } catch (Exception e) {
+            notEditableEquipments.add(equipment.getId());
+            equipmentNotModifiedCount += 1;
+            reports.add(ReportNode.newRootReportNode()
+                .withMessageTemplate(REPORT_KEY_EQUIPMENT_MODIFIED_REPORT_EXCEPTION,
+                        "        Cannot modify equipment ${" + VALUE_KEY_EQUIPMENT_NAME + "} : ${" + VALUE_KEY_ERROR_MESSAGE + "}")
+                .withUntypedValue(VALUE_KEY_EQUIPMENT_NAME, equipment.getId())
+                .withUntypedValue(VALUE_KEY_ERROR_MESSAGE, e.getMessage())
+                .withSeverity(TypedValue.TRACE_SEVERITY)
+                .build());
+            throw e;
+        }
+    }
+
+    private Map<UUID, String> getFilters() {
+        return getAssignmentInfosList().stream()
+            .flatMap(v -> v.getFilters().stream())
+            .filter(distinctByKey(FilterInfos::getId))
+            .collect(Collectors.toMap(FilterInfos::getId, FilterInfos::getName));
+    }
+
+    private void applyOnFilterEquipments(Network network,
+                                        Map<UUID, FilterEquipments> exportFilters,
```

Comment:
Why **export**Filters ? Where are they exported ?

Reply:
They are equipments which are matched to filters by requesting to filter library. 
I will rename to filterUuidEquipmentsMap

Reply:
DONE in https://github.com/gridsuite/network-modification-server/pull/526/commits/0177a47b3b62962723820a5b341da016038068c7